dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java

```diff
@@ -107,6 +107,13 @@ public static void received(Channel channel, Response response) {
         }
     }
 
+    public static void disconnected(Channel channel) {
```

Comment:
If the client connect to several server, DefaultFuture will keep all of the future which unfinished.
This method will disconnect all the futures which bind to different server?

Reply:
Maybe it will wake up a DefaultFuture that is bind to a server that didn't close ?

Reply:
Yes. Maybe we can check if future.channel==channel

Reply:
That is.
If not, the future will be waken up incorrect and return a incorrect value to user.

Reply:
If the future is waken up and the channel is still connected, the future will await again. We should adjust the timeout...

Reply:
Yeah, another problem.
:)